backend/studyagent/connectors/google.py

```
@@ -0,0 +1,263 @@
+    def __init__(self, token_secret: str, client=None) -> None:
+        from google.cloud import secretmanager
+        self._client = client or secretmanager.SecretManagerServiceClient()
+        self._token_secret = token_secret
+
+    def __call__(self, secret_ref: str):
+        from google.oauth2.credentials import Credentials
+        name = f"{self._token_secret}/versions/latest"
+        raw = self._client.access_secret_version(name=name).payload.data
+        data = json.loads(raw)
+        return Credentials(token=None, refresh_token=data["refresh_token"],
+                           token_uri="https://oauth2.googleapis.com/token", scopes=data["scopes"])
```

Ruff (0.16.2)

[error] Possible hardcoded password assigned to argument: "token_uri"

(S106)